feat(minifier): inline single use functions past sideeffectful expressions

sapphi-red opened the pull request:
Allowed literal functions to be inlined past sideeffectful expressions as it's in the same scope.
https://github.com/oxc-project/oxc/blob/b30bff56717c54028833a4a3cba0336be50ab60b/crates/oxc_ecmascript/src/constant_evaluation/is_literal_value.rs#L5-L18

close #13109

Boshen commented:
### Merge activity

* **Aug 30, 11:12 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

sapphi-red committed:
feat(minifier): inline single use functions past sideeffectful expressions (#13426)

Allowed literal functions to be inlined past sideeffectful expressions as it's in the same scope.
https://github.com/oxc-project/oxc/blob/b30bff56717c54028833a4a3cba0336be50ab60b/crates/oxc_ecmascript/src/constant_evaluation/is_literal_value.rs#L5-L18

close #13109
